File: Algorithms/DiagonalDifference/DiagonalDifference.scala
import scala.io._

object Solution {
    def main(args: Array[String]): Unit = {

        val n = StdIn.readLine.trim.toInt

        var diag1Sum = 0
        var diag2Sum = 0
        for (i <- 0 until n) {
            val arr = StdIn.readLine.trim().split(" ").map(_.toInt)
            for (j <- 0 until n) {
                if(i == j) {
                    diag1Sum += arr(j)
                    diag2Sum += arr(n-1-j)
                }
            }
        }
        println(Math.abs(diag1Sum-diag2Sum))
    }
}

File: Algorithms/DiagonalDifference/DiagonalDifference.cs
using System;
using System.Linq;

public class DiagonalDifference {


    public static int Main(string[] args)
    {

        var n = Convert.ToInt32(Console.ReadLine());
        int diag1Sum = 0;
        int diag2Sum = 0;
        for(int i = 0; i < n; i++) {
            int[] row = Console.ReadLine()
                        .Split(' ')
                        .Select(s => Convert.ToInt32(s))
                        .ToArray();
            for (int j = 0; j < n; j++)
            {
                if(i == j) {
                    diag1Sum += row[j];
                    diag2Sum += row[n-1-j];
                }
            }
        }
        Console.WriteLine(Math.Abs(diag1Sum-diag2Sum));
        return 0;
    }
}